src/comm.cuh

```diff
@@ -158,25 +165,24 @@ std::vector<comm_handle_t> recv_data_by_offset(
  * @returns                        Merged device buffer. The user of this function is responsible for freeing this
  *                                 returned buffer using RMM_FREE.
  */
-void* merge_free_received_offset(
-                                 std::vector<void *> received_data,
-                                 const std::vector<int64_t> &bucket_count,
-                                 size_t item_size,
-                                 int64_t &total_count,
-                                 Communicator *communicator=nullptr,
-                                 bool self_free=true)
+rmm::device_buffer
+merge_free_received_offset(
+    std::vector<void *> received_data,
```

Comment:
Passing a vector by value looks like a mistake to me.

Reply:
Fixed in https://github.com/rapidsai/distributed-join/pull/10/commits/dd3fe11b04d1bf08930ab7507ab8a10a6ce6f60c